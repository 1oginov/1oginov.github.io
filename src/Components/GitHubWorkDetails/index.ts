import GitHubWorkDetails from './GitHubWorkDetails';

export default GitHubWorkDetails;
