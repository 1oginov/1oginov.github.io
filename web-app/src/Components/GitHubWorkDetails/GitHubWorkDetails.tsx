import * as React from 'react';

import GitHub from 'Lib/GitHub';
import GitHubRepo from 'Lib/GitHub/GitHubRepo';
import GitHubTag from 'Lib/GitHub/GitHubTag';

interface Props {
  repo: GitHubRepo;
}

const GitHubWorkDetails: React.FunctionComponent<Props> = ({ repo }: Props) => {
  const [tags, setTags] = React.useState<GitHubTag[]>([]);

  React.useEffect(() => {
    GitHub.getTags(repo.name).then((t) => {
      setTags(t);
    });
  }, []);

  const version = tags.length > 0 ? tags[0].name : undefined;

  return (
    <>

      {repo.description && (
        <>
          {repo.description}
          <br />
        </>
      )}

      {repo.language && (
        <>
          <span title="Language">{repo.language}</span>
          {' '}
          /
          {' '}
        </>
      )}

      {repo.stargazers_count > 0 && (
        <>
          <span title="GitHub Stars">
            &#x2B50;
            {repo.stargazers_count}
          </span>
          {' '}
          /
          {' '}
        </>
      )}

      <span title="Last Updated">
        {new Date(repo.updated_at).toLocaleDateString(undefined, {
          day: 'numeric',
          month: 'short',
          year: 'numeric',
        })}
      </span>

      {version && (
        <>
          {' '}
          /
          {' '}
          <span title="Last Version">{version}</span>
        </>
      )}

    </>
  );
};

export default GitHubWorkDetails;
