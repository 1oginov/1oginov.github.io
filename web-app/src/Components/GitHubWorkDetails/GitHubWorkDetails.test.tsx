/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import mockData from 'Lib/GitHub/__fixtures__/data.json';

import { GitHubWorkDetails } from './GitHubWorkDetails';

it('matches snapshot', () => {
  const wrapper = shallow(
    <GitHubWorkDetails
      repo={mockData.repos[0]}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
