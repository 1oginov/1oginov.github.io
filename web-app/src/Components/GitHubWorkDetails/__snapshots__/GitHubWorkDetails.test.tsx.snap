// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  Clean but full featured AngularJS boilerplate using Gulp workflow and best practices
  <br />
  <span
    title="Language"
  >
    JavaScript
  </span>
   
  /
   
  <span
    title="GitHub Stars"
  >
    ‚≠ê
    29
  </span>
   
  /
   
  <span
    title="Last Updated"
  >
    Feb 16, 2021
  </span>
</Fragment>
`;
